tortoise/expressions.py

```diff
@@ -498,19 +552,28 @@ def resolve(self, resolve_context: ResolveContext) -> dict:
 
         default_values = self._resolve_default_values(resolve_context)
 
+        res = None
         if isinstance(self.field, str):
-            function = self._resolve_field_for_model(resolve_context, self.field)
-            function["field"] = self._get_function_field(function["field"], *default_values)
-            return function
-        elif isinstance(self.field, Function):
-            function = self.field.resolve(resolve_context)
-            function["field"] = self._get_function_field(function["field"], *default_values)
-            return function
-
-        field, field_object = F.resolver_arithmetic_expression(resolve_context.model, self.field)
-        if self.populate_field_object:
-            self.field_object = field_object
-        return {"joins": [], "field": self._get_function_field(field, *default_values)}
+            function_arg = self._resolve_nested_field(resolve_context, self.field)
+            term = self._get_function_field(function_arg.term, *default_values)
+            res = ResolveResult(
+                term=term,
+                joins=function_arg.joins,
+                output_field=function_arg.output_field,  # type: ignore
+            )
+        else:
+            function_arg = self.field.resolve(resolve_context)
+            term = self._get_function_field(function_arg.term, *default_values)
+            res = ResolveResult(
+                term=term,
+                joins=function_arg.joins,
+                output_field=function_arg.output_field,  # type: ignore
```

Comment:
Why you had to use so many type ignores?

Reply:
Great catch! This is something I never ran into before. Basically the `Field` class has [the following type checking code](https://github.com/tortoise/tortoise-orm/blob/develop/tortoise/fields/base.py#L162):

```
    # These methods are just to make IDE/Linters happy:
    if TYPE_CHECKING:

        def __new__(cls, *args: Any, **kwargs: Any) -> "Field[VALUE]":
            return super().__new__(cls)

        @overload
        def __get__(self, instance: None, owner: Type["Model"]) -> "Field[VALUE]": ...

        @overload
        def __get__(self, instance: "Model", owner: Type["Model"]) -> VALUE: ...

        def __get__(
            self, instance: Optional["Model"], owner: Type["Model"]
        ) -> "Field[VALUE] | VALUE": ...

        def __set__(self, instance: "Model", value: VALUE) -> None: ...
```

that makes it possible to have type hints for both the `Model` classes but also for `Model` instances. However, this seems to limit who can have the field attribute to only `Model` and its subclasses because of `instance` and `owner` arg typing. 

I added the `Field` property to `ResolveResult` to be able to use `Field` for type conversions and this is what caused warnings that are being ignored.

I tried to add overloaded versions of `__get__` to `Field` but this looks ugly because `Field` has to know about the `ResolveResult`:

```
@overload
        def __get__(self, instance: None, owner: Type["Model"]) -> "Field[VALUE]": ...

        @overload
        def __get__(self, instance: "Model", owner: Type["Model"]) -> VALUE: ...

        @overload
        def __get__(
            self, instance: ResolveResult, owner: Type["ResolveResult"]
        ) -> "Field[VALUE]": ...

        def __get__(
            self,
            instance: Union[Optional["Model"], ResolveResult],
            owner: Union[Type["Model"], Type[ResolveResult]],
        ) -> "Field[VALUE] | VALUE": ...
```

Please let me know if you have ideas on how to handle this better, thanks!